.circleci/config.yml

```diff
@@ -7,8 +17,182 @@ orbs:
+          docker run --rm -d \
+              --name taraxa-node-smoke-test-${DOCKER_BRANCH_TAG} \
+              --net smoke-test-net-${DOCKER_BRANCH_TAG} \
+              ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} \
+              single
+          mkdir -p  $PWD/test_build-d/
+          sleep 30
+          http_code=$(docker run --rm  -v $PWD/test_build-d:/data \
+                --net smoke-test-net-${DOCKER_BRANCH_TAG} \
+                byrnedo/alpine-curl \
+                -sS --fail -w '%{http_code}' -o /data/http.out \
+                --url taraxa-node-smoke-test-${DOCKER_BRANCH_TAG}:7777 \
+                -d '{
+                        "jsonrpc": "2.0",
+                        "id":"0",
+                        "method": "send_coin_transaction",
+                        "params":[{
+                            "nonce": 0,
+                            "value": 0,
+                            "gas": 0,
+                            "gas_price": 0,
+                            "receiver": "973ecb1c08c8eb5a7eaa0d3fd3aab7924f2838b0",
+                            "secret": "3800b2875669d9b2053c1aff9224ecfdc411423aac5b5a73d7a45ced1c3b9dcd"
+                        }]
+                    }')
+          cat $PWD/test_build-d/http.out || true
+          if [[ $http_code -eq 200 ]] ; then
+              exit 0
+          else
+              exit $http_code
+          fi
+      - run:
+         name: Cleanup smoke tests
+         command: |
+          docker kill taraxa-node-smoke-test-${DOCKER_BRANCH_TAG} || true
+          docker network rm smoke-test-net-${DOCKER_BRANCH_TAG} || true
+          docker images -a
+      - run:
+         name: Add Tags and push images
+         command: |
+           echo ${GCLOUD_SERVICE_KEY} | docker login -u _json_key --password-stdin https://gcr.io
+           docker tag ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} ${GCP_IMAGE}:${CIRCLE_SHA1}
+           docker tag ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} ${GCP_IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM}
+           docker tag ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} ${GCP_IMAGE}:${START_TIME}
+           docker tag ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} ${GCP_IMAGE}:${START_TIME}-${CIRCLE_SHA1}
+           docker tag ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} ${GCP_IMAGE}:latest
+           docker push ${GCP_IMAGE}:latest
+           docker push ${GCP_IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM}
+           docker push ${GCP_IMAGE}:${START_TIME}
+           docker push ${GCP_IMAGE}:${CIRCLE_SHA1}
+           docker push ${GCP_IMAGE}:${START_TIME}-${CIRCLE_SHA1}
+           if [[ ${PR} != "false" ]];then
+              docker tag ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} ${GCP_IMAGE}:pr-${PR}-${CIRCLE_BUILD_NUM} 
+              docker push ${GCP_IMAGE}:pr-${PR}-${CIRCLE_BUILD_NUM} 
+           fi
+           if [[ ${CIRCLE_BRANCH} == "master" ]];then
```

Comment:
We are also building the image `taraxa-node-develop` on GCR when merge to `develop` branch happens.
it's a different docker image at GCR, not only tags

Reply:
i may have not understood that's a different image, can you please point me out the difference so i can make it happen?

Reply:
We have those images on GCR
`taraxa-node`
`taraxa-node-develop`

It was asked that we had an image with the builds from `develop` branch, and only tags wouldn't fit the request
So `develop` branch builds and push this image. So the team knows if getting the latest from `taraxa-node-develop` means latest build from that branch.
We can change that, but current this is what we are doing at jenkins

Reply:
@sfrique  So to confirm taraxa-node images are only pushed when built on master branch?

Reply:
`taraxa-node` on docker hub yes.
we do push it to GCR for the PR build, but we do not tag it as the latest.
As said, this tagging did not get much thought, so we could change it if it would improve anything.

Reply:
Thank you , so now it is the current behaviour of the circleci pipeline,